crates/matrix-sdk-ui/tests/integration/timeline/pinned_event.rs

```diff
@@ -391,12 +376,86 @@ async fn test_pinned_timeline_with_no_pinned_events_and_an_utd_is_just_empty() {
     // returns an empty list
     let (items, _) = timeline.subscribe().await;
     assert!(items.is_empty());
+}
+
+#[async_test]
+async fn test_pinned_timeline_with_no_pinned_events_on_pagination_is_just_empty() {
+    let mut test_helper = TestHelper::new().await;
+    let room_id = test_helper.room_id.clone();
+    let event_id = event_id!("$1.localhost");
+    let sender_id = owned_user_id!("@example:localhost");
 
-    test_helper.server.reset().await;
+    // Join the room
+    let joined_room_builder = JoinedRoomBuilder::new(&room_id)
+        // Set up encryption
+        .add_state_event(StateTestEvent::Encryption);
+
+    // Sync the joined room
+    test_helper
+        .server
+        .mock_sync()
+        .ok_and_run(&test_helper.client, move |sync_builder| {
+            sync_builder.add_joined_room(joined_room_builder);
+        })
+        .await;
+
+    // Load initial timeline items: an empty `m.room.pinned_events` event
+    test_helper.sync_once(Vec::new(), Some(Vec::new())).await.expect("Sync failed");
+
+    let room = test_helper.client.get_room(&room_id).unwrap();
+    let pinned_timeline =
+        Timeline::builder(&room).with_focus(pinned_events_focus(1)).build().await.unwrap();
+
+    // The timeline couldn't load any events, but it expected none, so it just
+    // returns an empty list
+    let (pinned_items, mut pinned_events_stream) = pinned_timeline.subscribe().await;
+    assert!(pinned_items.is_empty());
+
+    // Create a non-pinned event to return in the pagination
+    let not_pinned_event = EventFactory::new()
+        .room(&room_id)
+        .sender(&sender_id)
+        .text_msg("Hey")
+        .event_id(event_id)
+        .into_raw_timeline();
+
+    test_helper
+        .server
+        .mock_room_messages()
+        .ok(assign!(RoomMessagesResponseTemplate::default(), {
+            chunk: vec![not_pinned_event.clone()]
```

Comment:
curious that clippy ain't found it…

```suggestion
            chunk: vec![not_pinned_event]
```

Reply:
It's probably because it was inside the macro. It tends to fail on those cases.